praw/models/reddit/emoji.py

```diff
@@ -162,8 +186,58 @@ def add(self, name, image_path):
             )
         response.raise_for_status()
 
+        data = {
+            "name": name,
+            "s3_key": upload_data["key"],
+            "post_flair_allowed": bool(post_flair),
+            "user_flair_allowed": bool(user_flair),
+            "mod_flair_only": bool(mod_only),
+        }
+
         url = API_PATH["emoji_upload"].format(subreddit=self.subreddit)
-        self._reddit.post(
-            url, data={"name": name, "s3_key": upload_data["key"]}
-        )
+        self._reddit.post(url, data=data)
         return Emoji(self._reddit, self.subreddit, name)
+
+    def update(
```

Comment:
The `update` method should belong to the `Emoji` class, because it operates on a single emoji, rather than to the `SubredditEmoji` (which represents all of a subreddit's emojis) class as it currently does.

Once this change is made, it's possible that code samples will need to be updated.

Reply:
The `add` method is also in the same class. What's different about it? (I ask because I'm learning, as that sounded argumentative)

Reply:
The reason that the methods belong in different classes is because they operate on slightly different things. When you add an emoji, you're creating a new emoji that doesn't yet exist and adding it to the collection of emojis that belong to a particular subreddit. Since the `SubredditEmoji` class represents all the emoji that a subreddit has, we use that.

In contrast, when updating an emoji, you're working with a specific emoji that already exists, which is what the `Emoji` class represents. For example, the `Emoji` class already has a `.delete()` method, which, like `update`, operates on a particular emoji. 

I hope this helps. Let me know if you need clarification.

Reply:
I apologize, I was positive I'd replied to this, even though we made contact shortly after anyway.

I think I've done this correctly. I've updated mine, corrected, clarified, etc.